saleor/graphql/order/types.py

```diff
@@ -48,10 +51,65 @@
 from .utils import validate_draft_order
 
 
+def get_order_discount_event(discount_obj: dict):
+    currency = discount_obj["currency"]
+
+    amount = prices.Money(Decimal(discount_obj["amount_value"]), currency)
+
+    old_amount = None
+    old_amount_value = discount_obj.get("old_amount_value")
+    if old_amount_value:
+        old_amount = prices.Money(Decimal(old_amount_value), currency)
+
+    return OrderEventDiscountObject(
+        value=discount_obj.get("value"),
+        amount=amount,
+        value_type=discount_obj.get("value_type"),
+        reason=discount_obj.get("reason"),
+        old_value_type=discount_obj.get("old_value_type"),
+        old_value=discount_obj.get("old_value"),
+        old_amount=old_amount,
+    )
+
+
+class OrderDiscount(graphene.ObjectType):
+    value_type = graphene.Field(
+        DiscountValueTypeEnum,
+        required=True,
+        description="Type of the discount: fixed or percent",
+    )
+    value = PositiveDecimal(
+        required=True,
+        description="Value of the discount. Can store fixed value or percent value",
+    )
+    reason = graphene.String(
+        required=False, description="Explanation for the applied discount."
+    )
+    amount = graphene.Field(Money, description="Returns amount of discount.")
+
+
+class OrderEventDiscountObject(OrderDiscount):
+    old_value_type = graphene.Field(
```

Comment:
Do we need these `old` prefixes?

Reply:
We need to store the normal values and the old values. The dashboard uses them to display the current value and the old value of the discount.